src/kubelet/metric/metric.go

```diff
@@ -214,38 +211,31 @@ func GroupStatsSummary(statsSummary v1.Summary) (definition.RawGroups, []error)
 		return g, errs
 	}
 
-PodListLoop:
 	for _, pod := range statsSummary.Pods {
 		rawPodMetrics, rawEntityID, err := fetchPodStats(pod)
 		if err != nil {
 			errs = append(errs, err)
-			continue PodListLoop
+			continue
 		}
+
 		g["pod"][rawEntityID] = rawPodMetrics
-	VolumeListLoop:
 		for _, volume := range pod.VolumeStats {
 			rawVolumeMetrics, err := fetchVolumeStats(volume)
 			if err != nil {
 				errs = append(errs, err)
-				continue VolumeListLoop
+				continue
```

Comment:
Does this change preserve the old behavior? Seems like an error fetching volume stats will cause the whole pod to be skipped without the labels.

Reply:
No, `continue` will skip the closest loop, in this case `for _, volume := range pod.VolumeStats {`, so yeah, it should be fine.

I can add tests for that.

Reply:
https://play.golang.org/p/_ezHOB-4wW3

Reply:
... You are entirely right, I don't know why I thought otherwise 😕